session/session.go

```diff
@@ -3240,12 +3240,16 @@ func (s *session) ShowProcess() *util.ProcessInfo {
 }
 
 // GetStartTSFromSession returns the startTS in the session `se`
-func GetStartTSFromSession(se interface{}) uint64 {
-	var startTS uint64
+func GetStartTSFromSession(se interface{}) (uint64, uint64) {
+	var startTS, processInfoID uint64
 	tmp, ok := se.(*session)
 	if !ok {
 		logutil.BgLogger().Error("GetStartTSFromSession failed, can't transform to session struct")
-		return 0
+		return 0, 0
+	}
+	processInfo := tmp.ShowProcess()
+	if processInfo != nil {
+		processInfoID = processInfo.ID
```

Comment:
we can get session id from tmp.sessionvars.id
the rest LGTM

Reply:
I think it's safer to distinguish the `processid` and `sessionid`, comparing the `processid` explicitly avoids the implicit constraint that they must be the same as the `analyze` process injects the `processid` at the start.